stellargraph/utils/loss.py

```diff
@@ -0,0 +1,27 @@
+import tensorflow as tf
+
+from ..core.experimental import experimental
+
+
+@experimental(reason="lack of unit tests")
+def graph_log_likelihood(y_true, y_pred):
+    '''
+    .. warning::
+
+        This function is experimental: it is insufficiently tested.
+    '''
+    batch_adj = tf.gather(y_true, [0], axis=1)
+
+    expected_walks = tf.gather(y_pred, [0], axis=1)
+    sigmoids = tf.gather(y_pred, [1], axis=1)
+
+    adj_mask = tf.cast((batch_adj == 0), "float32")
+
+    loss = tf.math.reduce_mean(
```

Comment:
I think matching the paper exactly would be:

```suggestion
    loss = tf.math.reduce_sum(
```

Could you say why you went for `reduce_mean`?

Reply:
oh yep, thats a bug!